OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ProviderAdminController.cs

```diff
@@ -253,4 +256,41 @@ public async Task<IActionResult> ManagedWorkshops()
 
         return Ok(relatedWorkshops);
     }
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Get ProviderAdmin by its id.
+    /// </summary>
+    /// <param name="providerAdminId">ProviderAdmin's id.</param>
+    /// <returns>Info about ProviderAdmin.</returns>
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ProviderAdminDto))]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet("{providerAdminId}")]
+    public async Task<IActionResult> GetProviderAdminById(string providerAdminId)
+    {
+        var providerAdmin = await providerAdminService.GetById(providerAdminId).ConfigureAwait(false);
+
+        if (providerAdmin == null)
+        {
+            return NoContent();
+        }
+
+        if (providerAdmin.IsDeputy)
+        {
+            await currentUserService.UserHasRights(new ProviderRights(providerAdmin.ProviderId))
```

Comment:
this check should be done in service layer methods

Reply:
We can't use CurrentUserService inside ProviderAdminService because we have an instance of ProviderAdminService in CurrentUserService, so I used currentUserService here.